tests/Unit/EmogrifierTest.php

```diff
@@ -690,6 +690,143 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
         static::assertContains(sprintf($expectedHtml, $cssDeclaration1, $cssDeclaration2), $result);
     }
 
+    /**
+     * @param string $selector A selector expression
+     * @param string $matchedTagPart Tag (in {@see EmogrifierTest::COMMON_TEST_HTML}) expected to be matched by the
+     *                               selector, without the closing '>', e.g. '<p class="p-1"'
+     *
+     * @return string[]
+     */
+    private function getMatchedCssDataSet($selector, $matchedTagPart)
+    {
+        // The sprintf placeholders %1$s and %2$s will automatically be replaced with CSS declarations
+        // like 'color: red;' or 'color: blue;'.
+        return [$selector . ' { %1$s }', $matchedTagPart . ' style="%1$s">'];
+    }
+
+    /**
+     * @param string $firstSelector A selector expression to be used in an earlier CSS declaration
+     * @param string $secondSelector Some other selector expression to be used in a later CSS declaration
+     * @param bool $firstSelectorIsMoreSpecific Whether the first selector's rules are expected to prevail
+     * @param string $matchedTagPart Tag (in {@see EmogrifierTest::COMMON_TEST_HTML}) expected to be matched by both
+     *                               selectors, without the closing '>', e.g. '<p class="p-1"'
+     *
+     * @return string[][]
+     */
+    private function getSelectorSpecificityCssDataSet(
+        $firstSelector,
+        $secondSelector,
+        $firstSelectorIsMoreSpecific,
```

Comment:
Instead of having this "toggle" parameter (that toggles between two different behaviors), we should have two separate, well-named methods.